src/Simplex/Chat.hs

```diff
@@ -835,7 +835,26 @@ processChatCommand = \case
             let aciContent = ACIContent SMDRcv $ CIRcvGroupInvitation ciGroupInv {status = CIGISAccepted} memRole
             updateDirectChatItemView userId ct itemId aciContent Nothing
           _ -> pure () -- prohibited
-  APIMemberRole _groupId _groupMemberId _memRole -> throwChatError $ CECommandError "unsupported"
+  APIMemberRole groupId memberId memRole -> withUser $ \user -> do
+    Group gInfo@GroupInfo {membership} members <- withStore $ \db -> getGroup db user groupId
+    if memberId == groupMemberId' membership
+      then changeMemberRole user gInfo members membership $ SGEUserRole memRole
+      else case find ((== memberId) . groupMemberId') members of
+        Nothing -> throwChatError CEGroupMemberNotFound
+        Just m -> changeMemberRole user gInfo members m $ SGEMemberRole memberId memRole
+    where
+      changeMemberRole user gInfo@GroupInfo {membership} members m@GroupMember {memberId = mId, memberRole = mRole, memberStatus = mStatus} gEvent = do
+        let userRole = memberRole (membership :: GroupMember)
+            canChangeRole = userRole >= GRAdmin && userRole >= mRole && userRole >= memRole && memberCurrent membership
+        unless canChangeRole $ throwChatError CEGroupUserRole
+        withChatLock . procCmd $ do
+          unless (mRole == memRole) $ do
+            withStore' $ \db -> updateGroupMemberRole db user m memRole
+            unless (mStatus == GSMemInvited) $ do
```

Comment:
How does invited member learn his role has changed? Maybe it should be prohibited - a new invitation has to be sent, maybe even as part of this command? I think repeat invitations may already treat this case, better check though

Reply:
updated